misc/agent-introspection-validator/agent-introspection-validator.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+)
+
+const (
+	agentIntrospectionEndpoint = "http://localhost:51678/v1/"
+	maxRetries                 = 4
+	durationBetweenRetries     = time.Second
+	containerName              = "agent-introspection-validator"
+	tasksMetadataRespType      = "tasks metadata"
+	taskMetadataRespType       = "task metadata"
+)
+
+// TaskResponse is the schema for the task response JSON object
+type TaskResponse struct {
+	Arn           string
+	DesiredStatus string `json:",omitempty"`
+	KnownStatus   string
+	Family        string
+	Version       string
+	Containers    []ContainerResponse
+}
+
+// TasksResponse is the schema for the tasks response JSON object
+type TasksResponse struct {
+	Tasks []*TaskResponse
+}
+
+// ContainerResponse is the schema for the container response JSON object
+type ContainerResponse struct {
+	DockerID   string
```

Comment:
just making a note since i asked you offline, this should be `DockerId` to match the original API.

Reply:
Thanks! I was planning to update it when the fix PR is rebased, will add a TODO in the PR description.